multicluster/controllers/multicluster/member/resourceimport_controller_test.go

```diff
@@ -501,3 +507,95 @@ func TestResourceImportReconciler_handleUpdateEvent(t *testing.T) {
+	for i := 0; i < common.DefaultWorkerCount; i++ {
+		go wait.Until(fm.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (fm *fakeManager) worker() {
+	for fm.syncNextItemInQueue() {
+	}
+}
+
+func (fm *fakeManager) syncNextItemInQueue() bool {
+	resImpObj, quit := fm.queue.Get()
+	if quit {
+		return false
+	}
+	defer fm.queue.Done(resImpObj)
+	resImp := resImpObj.(*mcsv1alpha1.ResourceImport)
+	err := fm.remoteClient.Create(ctx, resImp)
+	if err != nil {
+		fm.queue.AddRateLimited(resImp)
+		return true
+	}
+	// Simulate ResourceImport create event and have LabelIdentityResourceImportReconciler reconcile it.
+	req := ctrl.Request{NamespacedName: types.NamespacedName{Namespace: resImp.Namespace, Name: resImp.Name}}
+	if _, err = fm.reconciler.Reconcile(ctx, req); err != nil {
+		fm.queue.AddRateLimited(resImp)
+		return true
+	}
+	fm.queue.Forget(resImp)
+	return true
+}
+
+func TestStaleControllerNoRaceWithResourceImportReconciler(t *testing.T) {
+	fakeClient := fake.NewClientBuilder().WithScheme(common.TestScheme).WithLists().Build()
+	fakeRemoteClient := fake.NewClientBuilder().WithScheme(common.TestScheme).WithLists().Build()
+	ca := commonarea.NewFakeRemoteCommonArea(fakeRemoteClient, "leader-cluster", common.LocalClusterID, "antrea-mcs", nil)
+
+	mcReconciler := NewMemberClusterSetReconciler(fakeClient, common.TestScheme, "default", true)
+	mcReconciler.SetRemoteCommonArea(ca)
+	c := multicluster.NewStaleResCleanupController(fakeClient, common.TestScheme, "default", mcReconciler, multicluster.MemberCluster)
+	r := newLabelIdentityResourceImportReconciler(fakeClient, scheme, fakeClient, localClusterID, "default", ca)
+
+	stopCh := make(chan struct{})
+	q := workqueue.NewRateLimitingQueue(workqueue.DefaultItemBasedRateLimiter())
+	numInitialResImp := 50
+	for i := 1; i <= numInitialResImp; i++ {
+		resImp := &mcsv1alpha1.ResourceImport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "label-identity-" + strconv.Itoa(i),
+				Namespace: "antrea-mcs",
+			},
+			Spec: mcsv1alpha1.ResourceImportSpec{
+				LabelIdentity: &mcsv1alpha1.LabelIdentitySpec{
+					Label: "ns:kubernetes.io/metadata.name=ns&pod:seq=" + strconv.Itoa(i),
+					ID:    uint32(i),
+				},
+			},
+		}
+		q.Add(resImp)
+	}
+	mgr := fakeManager{
+		reconciler:   r,
+		remoteClient: fakeRemoteClient,
+		queue:        q,
+	}
+	// Give the fakeManager a head start. LabelIdentityResourceImportReconciler should be busy
+	// reconciling all new ResourceImport events.
+	go mgr.Run(stopCh)
+	// The staleController should not erroneously delete any LabelIdentities while the reconciliation
+	// of newly added ResourceImports are in-flight.
+	go c.Run(stopCh)
+	time.Sleep(1 * time.Second)
+	actLabelIdentities := &mcsv1alpha1.LabelIdentityList{}
+	err := fakeClient.List(ctx, actLabelIdentities)
+	assert.NoError(t, err)
+	// Verify that no label identities are deleted as part of the cleanup.
+	assert.Equal(t, numInitialResImp, len(actLabelIdentities.Items))
```

Comment:
I suppose you should stop the go routine with `<-stopCh`?

Reply:
I don't think we need to. Adding `<-stopCh` puts the testcase in a infinite loop.

Reply:
OK, I think it should be `defer close(stopCh)` after line `stopCh := make(chan struct{})`.

Reply:
Done